src/main/java/frc/robot/commands/AutonomousDistance.java

```diff
@@ -0,0 +1,22 @@
+// Copyright (c) FIRST and other WPILib contributors.
+// Open Source Software; you can modify and/or share it under the terms of
+// the WPILib BSD license file in the root directory of this project.
+
+package frc.robot.commands;
+
+import edu.wpi.first.wpilibj2.command.SequentialCommandGroup;
+import frc.robot.subsystems.drive.DriveSubsystemXrp;
+
+public class AutonomousDistance extends SequentialCommandGroup {
+  /**
+   * Creates a new Autonomous Drive based on distance. This will drive out for a specified distance,
+   * turn around and drive back.
+   *
+   * @param drivetrain The drivetrain subsystem on which this command will run
+   */
+  public AutonomousDistance(DriveSubsystemXrp drivetrain) {
```

Comment:
I would change the "DriveSystemXrp" (the _implementation_) to "Drive" (the _interface_), so that this command could be re-used in **any** subsystem that implements the Drive interface.  E.g. we could have the same command work on Romi, XRP, and even the actual comp bot.

Reply:
How would reccomend I do this?  I would have to add  a function to the Drive interface for getting the left and right positions. In the current code I do this by accessing that data through the DriveSubsystemXRP through the drivedifferentialIOXrp/DriveDifferentialIo class/interface but that is really specific to robots with a left and right wheels.  Adding that logic to the Drive interface would limit it and all derived implementations to 2wd bots.